### PR TITLE
Add rule notes to license match results

### LINKED ISSUE #1377: Include license rule notes in data returned with a license match
There is valuable information in these notes and they provide context for some less common rules and why these would have been detected.

### PR DESCRIPTION
**Pull request by n1khilmane**
Fixes #1377 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

Signed-off-by: Nikhil Mane <nikhilmane1919@gmail.com>

### CODE REVIEW

**Comment by Pratikrocks** [comment]
@n1khilmane I think you should create a new branch from `develop` as this PR contains commits from your previous `branches`

File: snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+confinement: devmode # use 'strict' once you have the right plugs and slots
+
+parts:
+  scancode-toolkit:
+    # See 'snapcraft plugins'
+    plugin: python
+    python-version: python2
+    source: .
+    stage-packages: 
+       - python-six
```
Comment (Pratikrocks) [inline snapcraft.yaml:17]: I think this is not related to this PR

**Comment by n1khilmane** [comment]
@Pratikrocks Thanks mate!!, I clearly missed on that , my bad!!

File: snapcraft.yaml (hunk 1 above)
Comment (n1khilmane) [reply snapcraft.yaml:17]: yes, you are right

Comment (n1khilmane) [reply snapcraft.yaml:17]: i am facing some issues, what should be done to resolve this?

Comment (Pratikrocks) [reply snapcraft.yaml:17]: What kind of issues ?

Comment (n1khilmane) [reply snapcraft.yaml:17]: should i close this PR and create a new one?

Comment (Pratikrocks) [reply snapcraft.yaml:17]: Yeah that can also be done
